apache/cloudstack
# Add support to RBD erasure code pools

### Description

This PR adds support for Ceph erasure code pools, allowing users to specify the data pool required for using this type of Ceph pool.

### Types of changes


### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale


### Screenshots (if appropriate):
Details of the data pool in the UI

![image](https://github.com/user-attachments/assets/4e1e5790-972d-403e-b2a3-c69c8a61e3e1)

### How Has This Been Tested?

The following tests were executed to verify this patch; tests were executed on replicated pools as well to check if this patch would interfere in it.

- Deploy of VM in a Ceph replicated pool
- Deploy of VM in a Ceph erasure code pool
- Volume resize in a Ceph replicated pool
- Volume resize in a Ceph erasure code pool
- Live migration of VM with volume allocated in a Ceph replicated pool
- Live migration of VM with volume allocated in a Ceph erasure code pool
- Attach, mount and utilization of volume in a Ceph replicated pool
- Attach, mount and utilization of volume in a Ceph erasure code pool
- Template creation from a volume in a Ceph replicated pool
- Deploy of VM from the template based on volume in a Ceph replicated pool
- Template creation from a volume in a Ceph erasure code pool
- Deploy of VM from the template based on volume in a Ceph erasure code pool
- Volume migration from/to Ceph replicated pool to/from NFS
- Volume migration from/to Ceph replicated pool to/from iSCSI (SharedMountPoint)
- Volume migration from/to Ceph erasure code pool to/from NFS
- Volume migration from/to Ceph erasure code pool to/from iSCSI (SharedMountPoint)
- Volume migration from/to Ceph erasure code pool to/from Ceph replicated pool
- Snapshot of a volume from Ceph replicated pool
- Revert snapshot of a volume from Ceph replicated pool
- Snapshot of a volume from Ceph erasure code pool
- Revert snapshot of a volume from Ceph erasure code pool


#### How did you try to break this feature and the system with this change?

I tried to migrate volumes from different type of pool types (NFS, iSCSI and RBD) to each other, verifying that the migration was successful.
